com.unity.template.renderstreaming-hd/Assets/Scripts/RenderStreaming.cs

```diff
@@ -196,19 +214,70 @@ void OnIceCandidate(ISignaling signaling, CandidateData e)
 
         void OnDataChannel(RTCPeerConnection pc, RTCDataChannel channel)
         {
-            Dictionary<int, RTCDataChannel> channels;
-            if (!mapChannels.TryGetValue(pc, out channels))
+            if (!mapChannels.TryGetValue(pc, out var channels))
             {
                 channels = new Dictionary<int, RTCDataChannel>();
                 mapChannels.Add(pc, channels);
             }
             channels.Add(channel.Id, channel);
 
-            if(channel.Label == "data")
+            if (channel.Label != "data")
             {
-                channel.OnMessage = new DelegateOnMessage(bytes => { RemoteInput.ProcessInput(bytes); });
-                channel.OnClose = new DelegateOnClose(() => { RemoteInput.Reset(); });
+                return;
             }
+
+            RemoteInput input = RemoteInputReceiver.Create();
+
+            // device.current must be changed after creating devices
+            m_defaultInput.MakeCurrent();
+
+            m_channelIdAndRemoteInput.Add(channel, input);
+            channel.OnMessage = bytes => m_channelIdAndRemoteInput[channel].ProcessInput(bytes);
+            channel.OnClose = () => OnCloseChannel(channel);
+
+            SimpleCameraController controller = m_listController
```

Comment:
How often is this function called ?
A bit curious because we are putting linq here, and it might slow the program down if this function is called many times.

Reply:
This function called when users connect with the Unity instance from the browser.
Instead of putting linq, it can be better to make another collection to identify the controller assigned input.

Reply:
Ok, if it's not called that often, that I think it's find to use linq.
Making another collection might be a better idea. I'll leave it up to you.